### ISSUE TITLE
Expected type error on delegate

### ISSUE THREAD

**Issue by auduchinok** [dotnet/fsharp#5607]
```
type Foo = delegate of unit -> unit 

let f1 = Foo(ignore)
f1.Invoke()

Foo(ignore).Invoke()
```

```
  Library.fs(9, 1): [FS0001] This expression was expected to have type    'Foo1'    but here has type    'unit'
```

<img width="299" alt="screen shot 2018-09-07 at 16 59 43" src="https://user-images.githubusercontent.com/3923587/45223281-83879a80-b2bf-11e8-8352-96d98b0a0644.png">

**Comment by cartermp** [comment]
Wrapping it all in parens fixes it, interestingly:

`(Foo(ignore)).Invoke()`

**Comment by matthid** [comment]
![image](https://user-images.githubusercontent.com/1866463/45230258-9eaad800-b2c8-11e8-96ba-34e81ce19996.png)

So it seems the underline is also not correct (ie the compiler is probably complaining about some sub-expression)?

**Comment by dsyme** [comment]
@auduchinok I finally fixed this, see #12920 .
